secure-ios-app/services/DevicetrustService.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  DevicetrustService.swift
+//  secure-ios-app
+//
+//  Created by Tom Jackman on 30/11/2017.
+//  Copyright © 2017 Wei Li. All rights reserved.
+//
+
+import Foundation
+import LocalAuthentication
+import DTTJailbreakDetection
+
+protocol DeviceTrustService {
+    func performTrustChecks() -> [Detector]
+}
+
+class iosDeviceTrustService: DeviceTrustService {
+    
+    var detections: [Detector]
+    // Detections
+    // Positive: Detection occured that is deemed a security issue
+    // Negative: No threat found for the detection
+    let DETECTION_DEVICE_LOCK_POSITIVE = "Device Lock Not Set"
+    let DETECTION_DEVICE_LOCK_NEGATIVE = "Device Lock Set"
+    let JAILBREAK_DETECTED_POSITIVE = "Jailbreak Detected"
+    let JAILBREAK_DETECTED_NEGATIVE = "Jailbreak Not Detected"
+    let DEBUG_MODE_DETECTED_POSITIVE = "Debug Mode Detected"
+    let DEBUG_MODE_DETECTED_NEGATIVE = "Debug Mode Not Detected"
+    let EMULATOR_DETECTED_POSITIVE = "Emulator Access Detected"
+    let EMULATOR_DETECTED_NEGATIVE = "Physical Device Access Detected"
+    let LATEST_OS_DETECTED_POSITIVE = "Latest OS Version Not Installed"
+    let LATEST_OS_DETECTED_NEGATIVE = "Latest OS Version Installed"
+    
+    /**
+     - Initilise the iOS Device Trust Service
+    */
+    init() {
+        self.detections = []
+    }
+    
+    /**
+     - Perform the Device Trust Checks
+     
+     - Returns: A list of Detector objects
+     */
+    func performTrustChecks() -> [Detector] {
+        detectDeviceLock()
+        detectJailbreak()
+        detectEmulator()
+        detectDebugabble()
+        detectLatestOS()
+        
+        return self.detections
```

Comment:
this is fine for our app, but the down side is that the individual check call not be used on its own. A slightly better approach here is to make each function return a `Detector` instance and then in `performTrustChecks` add the returned detectors to the global detections array. This will allow each check to be called on its own.

WDYT?